src/libopensc/reader-pcsc.c

```diff
@@ -253,8 +253,8 @@ static int pcsc_internal_transmit(sc_reader_t *reader,
 	sRecvPci.dwProtocol = opensc_proto_to_pcsc(reader->active_protocol);
 	sRecvPci.cbPciLength = sizeof(sRecvPci);
 
-	dwSendLength = sendsize;
-	dwRecvLength = *recvsize;
+	dwSendLength = (DWORD)sendsize;
+	dwRecvLength = (DWORD)*recvsize;
```

Comment:
In theory, converting the size_t (8B on x86_64), while `DWORD` is `uint32_t` (4B) if I see right. Most of the callers call this with much smaller sizes though (i did not check all though). If we know that the buffers are always smaller, I think it should have a comment describing this. If it could be larger than uin32_t max, it should have sanity check that the conversion wont overflow.

Reply:
Maybe modify function signature to let callers to know?

Reply:
That works ok for the recvsize, but I feel like it just moves the problem elsewhere where you need to do the casts to compare with some other types or return the lengths. But still looks reasonable.